riocli/device/model.py

```diff
@@ -36,7 +46,21 @@ def find_object(self, client: Client) -> bool:
         return obj
 
     def create_object(self, client: Client, **kwargs) -> v1Device:
-        device = client.create_device(self.to_v1())
+        if not self.spec.get('virtual', {}).get('enabled', False):
+            device = client.create_device(self.to_v1())
+            return device
+        hwil_device = self.create_hwil_device()
```

Comment:
nit: leave a blank line here.
```suggestion
        if not self.spec.get('virtual', {}).get('enabled', False):
            device = client.create_device(self.to_v1())
            return device
            
        hwil_device = self.create_hwil_device()
```